test/js/test.js

```diff
@@ -445,7 +373,7 @@ function getReadyToValidate(err, resp, body) {
   function httpResponder(req, response) {
     console.log("\nGot request for", req.url);
     var host = req.headers["host"];
-    if ((host === state.domain || /localhost/.test(state.newRegistrationURL)) &&
+    if ((host.split(/:/)[0] === state.domain || /localhost/.test(state.newRegistrationURL)) &&
```

Comment:
Good point! But if somebody tries to contact this validation server with a Host header equal to an IPv6 address,  we are happy to fail. This check is just making sure that the CA is actually asking for the hostname we wanted, which should never by an IP address of any sort (the CA would reject it).